File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcGroup/IfcStructuralResultGroup/IfcStructuralResultGroup.cs
public class IfcStructuralResultGroup : IfcGroup
{
    public IfcAnalysisTheoryTypeEnum TheoryType { get; set; }
    public IfcStructuralLoadGroup ResultForLoadGroup { get; set; }
    public IfcBoolean IsLinear { get; set; }

    // === INVERSE CLAUSES ===
    // ResultGroupFor : SET [0:1] OF IfcStructuralAnalysisModel FOR HasResults

    // === WHERE CLAUSES ===
    // HasObjectType : (TheoryType <> IfcAnalysisTheoryTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType)
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcGroup/IfcStructuralResultGroup/IfcStructuralResultGroup.vb
Public Class IfcStructuralResultGroup Inherits IfcGroup
    Public TheoryType As IfcAnalysisTheoryTypeEnum
    Public ResultForLoadGroup As IfcStructuralLoadGroup
    Public IsLinear As IfcBoolean

    ' === INVERSE CLAUSES ===
    ' ResultGroupFor : SET [0:1] OF IfcStructuralAnalysisModel FOR HasResults

    ' === WHERE CLAUSES ===
    ' HasObjectType : (TheoryType <> IfcAnalysisTheoryTypeEnum.USERDEFINED) OR EXISTS(SELF\IfcObject.ObjectType)
End Class
